cmd/podman/artifact/inspect.go

```diff
@@ -26,25 +38,63 @@ func init() {
 		Parent:  artifactCmd,
 	})
 
-	// TODO When things firm up on inspect looks, we can do a format implementation
-	// flags := inspectCmd.Flags()
-	// formatFlagName := "format"
-	// flags.StringVar(&inspectFlag.format, formatFlagName, "", "Format volume output using JSON or a Go template")
-
-	// This is something we wanted to do but did not seem important enough for initial PR
-	// remoteFlagName := "remote"
-	// flags.BoolVar(&inspectFlag.remote, remoteFlagName, false, "Inspect the image on a container image registry")
-
-	// TODO When the inspect structure has been defined, we need to uncomment and redirect this.  Reminder, this
-	// will also need to be reflected in the podman-artifact-inspect man page
-	// _ = inspectCmd.RegisterFlagCompletionFunc(formatFlagName, common.AutocompleteFormat(&machine.InspectInfo{}))
+	flags := inspectCmd.Flags()
+	formatFlagName := "format"
+	flags.StringVarP(&inspectOpts.Format, formatFlagName, "f", "json", "Format the output to a Go template or json")
+	_ = inspectCmd.RegisterFlagCompletionFunc(formatFlagName, common.AutocompleteFormat(&artifactInspectFormat{}))
 }
 
 func inspect(_ *cobra.Command, args []string) error {
 	artifactOptions := entities.ArtifactInspectOptions{}
-	inspectData, err := registry.ImageEngine().ArtifactInspect(registry.Context(), args[0], artifactOptions)
+
+	// Collect all inspect results
+	inspectData := make([]any, 0, len(args))
+	errs := []error{}
+
+	for _, name := range args {
+		data, err := registry.ImageEngine().ArtifactInspect(registry.Context(), name, artifactOptions)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		inspectData = append(inspectData, data)
+	}
+
+	// If all artifacts failed to inspect, return the errors
+	if len(inspectData) == 0 && len(errs) > 0 {
+		return errs[0]
+	}
+
+	// Always print an empty array if no data
+	if inspectData == nil {
+		inspectData = []any{}
+	}
+
+	// Handle output formatting
+	var err error
+	switch {
+	case report.IsJSON(inspectOpts.Format) || inspectOpts.Format == "":
+		err = utils.PrintGenericJSON(inspectData)
+	default:
+		// User has given a custom --format
+		var rpt *report.Formatter
+		rpt, err = report.New(os.Stdout, "inspect").Parse(report.OriginUser, inspectOpts.Format)
+		if err != nil {
+			return err
+		}
+		defer rpt.Flush()
+
+		err = rpt.Execute(inspectData)
+	}
+
 	if err != nil {
 		return err
 	}
-	return utils.PrintGenericJSON(inspectData)
+
+	// Return first error if any occurred (but after printing successful results)
+	if len(errs) > 0 {
```

Comment:
This doesn't look right. We should be printing all the errors except the last one and returning the last one.